runtime/v2/binary.go

```diff
@@ -77,13 +77,20 @@ func (b *binary) Start(ctx context.Context, opts *types.Any, onClose func()) (_
 	}
 	// Windows needs a namespace when openShimLog
 	ns, _ := namespaces.Namespace(ctx)
-	f, err := openShimLog(namespaces.WithNamespace(context.Background(), ns), b.bundle, client.AnonDialer)
+	shimCtx, cancelShimLog := context.WithCancel(namespaces.WithNamespace(context.Background(), ns))
```

Comment:
I would feel safer if we just add a:
```go
defer func() {
    if err != nil {
        cancelShimLog()
}
```
right here. Because for example why is the error return path from client.Connect not issuing a `cancelShimLog` but all other error paths are? It seems easy to get this one wrong so doing it on all error paths via `defer` seems better

Reply:
@jterry75 thanks! updated